org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/DatetimeBox.java

```diff
@@ -38,6 +41,9 @@ public DatetimeBox() {
 		dateBox.setCols(10);
 		timeBox = new Timebox();
 		timeBox.setCols(10);
+		MLanguage mlang = MLanguage.get(Env.getCtx(), Language.getLoginLanguage());
+		if (mlang != null && mlang.get_ID() > 0 && mlang.getTimePattern() != null)
+			timeBox.setFormat(mlang.getTimePattern());

```

Comment:
Since we are already doing this in the DateBox constructor above, I think it would be more consistent to do this in the Timebox constructor as well.